staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -220,14 +221,20 @@ func (c *AvailableConditionController) sync(key string) error {
 			availableCondition.Reason = "EndpointsNotFound"
 			availableCondition.Message = fmt.Sprintf("cannot find endpoints for service/%s in %q", apiService.Spec.Service.Name, apiService.Spec.Service.Namespace)
 			apiregistration.SetAPIServiceCondition(apiService, availableCondition)
-			_, err := updateAPIServiceStatus(c.apiServiceClient, originalAPIService, apiService)
+			if _, err2 := updateAPIServiceStatus(c.apiServiceClient, originalAPIService, apiService); err2 != nil {
+				klog.Errorf("cannot update API service status: %v", err2)
+				return err2
```

Comment:
Not clear to me that the failure to update the api service status is the most useful error to return here. The status change we are trying to set is that the endpoint is missing. If we can't set that error message then this error message is now the best chance of notifying someone of the missing endpoint. I think the log message that the update failed is a good change. However I think we should still be returning err.

Reply:
@cheftako 
I have addressed your comment above

Reply:
I can buy that we were returning the wrong error. I am somewhat unsettled that changing this did not invalidate some test.